compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/collectors/components/ErrorNodeDiagnosticCollectorComponent.kt

```diff
@@ -39,9 +39,9 @@ class ErrorNodeDiagnosticCollectorComponent(
     }
 
     override fun visitResolvedTypeRef(resolvedTypeRef: FirResolvedTypeRef, data: CheckerContext) {
-        val errorType = resolvedTypeRef.type as? ConeClassErrorType ?: return
-        val source = resolvedTypeRef.source ?: return
-        reportFirDiagnostic(errorType.diagnostic, source, reporter, data)
+        assert(resolvedTypeRef.type !is ConeKotlinErrorType) {
```

Comment:
I use this assertion to detect remaining cases, but let me know if you think it's better to remove this override as a whole.